### PR TITLE
added sample code for armclient  in China azure.

### LINKED ISSUE #45222: [QUERY]Need to get a ArmClient construction sample for Azure in China
### Library name and version

Azure.ResourceManager

### Query/Question

Need to get a ArmClient construction sample for Azure in China

### Environment

_No response_

### PR DESCRIPTION
**Pull request by welovej**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/resourcemanager/Azure.ResourceManager/README.md [hunk 1]
```diff
@@ -53,6 +53,25 @@ using Azure.ResourceManager.Resources;
 ArmClient client = new ArmClient(new DefaultAzureCredential());
 ```
 
+Note: if you want to authenticate with the azure in China, you can use the following code:
+```
```
Comment (jsquire) [inline sdk/resourcemanager/Azure.ResourceManager/README.md:57]: This must be a snippet in order to ensure that it can be compiled and does not rot.

**Review by jsquire — DISMISSED** [review]

File: sdk/resourcemanager/Azure.ResourceManager/README.md (hunk 1 above)
Comment (welovej) [reply sdk/resourcemanager/Azure.ResourceManager/README.md:71]: Fixed

File: sdk/resourcemanager/Azure.ResourceManager/samples/DocSamples/Readme.cs [hunk 2]
```diff
@@ -24,6 +24,26 @@ public void ClientAuth()
...
+            string tenantId = "your-tenant-id";
+            string clientId = "your-client-id";
+            string clientSecret = "your-client-secret";
+            string subscriptionId = "your-subscription-id";
+            //ArmClientOptions to set the Azure China environment
+            var armOptions = new ArmClientOptions { Environment = ArmEnvironment.AzureChina };
+            // AzureAuthorityHosts to set the Azure China environment
+            var authorityHost = AzureAuthorityHosts.AzureChina;
+            // Create ClientSecretCredential for authentication
+            var credential = new ClientSecretCredential(tenantId, clientId, clientSecret, new TokenCredentialOptions { AuthorityHost = authorityHost });
```
Comment (ArcturusZhang) [inline sdk/resourcemanager/Azure.ResourceManager/samples/DocSamples/Readme.cs:42]: in general, in sample codes, we should avoid using `var`. Could you change these `var` here to their explicit type?
The type of `credential` here could be `TokenCredential`.

**Review by ArcturusZhang — APPROVED** [review]

**Comment by welovej** [comment]
Hi @jsquire, please help to review.

**Comment by jsquire** [comment]
> Hi @jsquire, please help to review.

I've cleared my blocker now that the snippet is in place.

**Pull request merged**